osu.Game/Screens/Edit/Editor.cs

```diff
@@ -519,7 +544,7 @@ public override bool OnExiting(IScreen next)
             // To update the game-wide beatmap with any changes, perform a re-fetch on exit.
             // This is required as the editor makes its local changes via EditorBeatmap
             // (which are not propagated outwards to a potentially cached WorkingBeatmap).
-            var refetchedBeatmap = beatmapManager.GetWorkingBeatmap(Beatmap.Value.BeatmapInfo);
+            var refetchedBeatmap = beatmapManager.GetWorkingBeatmap(editorBeatmap.BeatmapInfo);
```

Comment:
The reason this is changed is that `GetWorkingBeatmap()` silently stores back the provided `BeatmapInfo` to the `WorkingBeatmap`. And because the beatmapset is refetched anew from the db context in BDL, the old code would pass a stale `BeatmapInfo` reference to the working beatmap, leading to things like the difficulty name updating on the carousel in song select but not the beatmap wedge on the left.